HW1/P2/P2b.py

```diff
@@ -1,3 +1,27 @@
 from P2 import *
 
 # Your code here
+from pyspark import SparkContext
+import matplotlib.pyplot as plt
+import numpy as np
+
+sc = SparkContext("local[4]", "HW1-2b Default Partioning")
+sc.setLogLevel("ERROR")
+
+#set up 2000x2000 array and transform array into [-2,2]x[-2,2] grid... 
+rdd = sc.parallelize(xrange(0,2001),10)
+grid = rdd.cartesian(rdd).repartition(100) #Creates ordered pairs, redistribute randomly...
```

Comment:
I'm surprised at how nonuniform the distribution of compute is when using the repartition function. It's clearly a better strategy than using the native partitioning scheme, but a good randomization scheme should in theory lead to a more uniformly distributed compute. I would have thought repartition would have randomized things together better; perhaps repartitioning to 1 partition and then back to 100 would be a better "randomization". 

E.g., I used a partitionBy() using a random.randint function, and I was able to get a more uniform distribution.